pkg/runtime-tools/generate/generate.go

```diff
@@ -309,6 +310,14 @@ func (g *Generator) AdjustCgroupsPath(path string) {
 	}
 }
 
+// AdjustOomScoreAdj adjusts the kernel's Out-Of-Memory (OOM) killer score for the container.
+// This may override kubelet's settings for OOM score.
+func (g *Generator) AdjustOomScoreAdj(score *nri.OptionalInt) {
+	if score != nil {
+		g.SetProcessOOMScoreAdj(int(score.Value))
+	}
```

Comment:
Thanks for the suggestion! After we set OomScoreAdj to nil, We can handle the nil here.
I try to call the func g.initConfigProcess() by calling func g.SetProcessOOMScoreAdj(0), and then set g.Config.Process.OOMScoreAdj to nil.

```suggestion
	} else {
		g.SetProcessOOMScoreAdj(0)
		g.Config.Process.OOMScoreAdj = nil
	}
```

Reply:
sgtm  let's try it / test it out

Reply:
We tested it with the following code
```	
oomScoreAdj:= 555
adjustment.SetLinuxOomScoreAdj(&oomScoreAdj)
adjustment.SetLinuxOomScoreAdj(nil)
```
The second set unset the oomscoreadj with the api. Finally, the oom_score_adj is set to 996 by kubelet.

> cat /proc/2213925/oom_score_adj
996
This is work! @mikebrow

Reply:
This code path appears to be the source of https://github.com/containerd/nri/issues/117.  @mikebrow, do we need to support a plugin both setting an explicit adjustment and then clearing its own adjustment?

Reply:
nod.. while I think it's a valid path to unset given 0 / 999 etc.. is not unset.. we need to ensure we are not accidentally unsetting a valet oom from kubelet. @klihub FYI

Reply:
manual:  

``` OOMScoreAdjust=
           Sets the adjustment value for the Linux kernel's
           Out-Of-Memory (OOM) killer score for executed processes.
           Takes an integer between -1000 (to disable OOM killing of
           processes of this unit) and 1000 (to make killing of
           processes of this unit under memory pressure very likely).
           See The /proc Filesystem[6] for details. If not specified
           defaults to the OOM score adjustment level of the service
           manager itself, which is normally at 0.
```
note behavior normally 0 when unset, but is configurable making unset a unique state